plugins/nf-azure/src/main/nextflow/cloud/azure/nio/AzPath.groovy

```diff
@@ -319,14 +320,16 @@ class AzPath implements Path {
 
 
     String toUriString() {
+        final accountName = AzConfig.getConfig().storage().accountName
         if( path.isAbsolute() ) {
-            return "${AzFileSystemProvider.SCHEME}:/${path.toString()}"
+            return "/${accountName}${path.toString()}"
```

Comment:
Should there not still be the `az://` prefix here? Or you're saying that Azure will mount the files at `/<accountName>/...` ?

Reply:
Azure-Tes will mount these files

Reply:
This will break everything

Reply:
What do you mean @pditommaso ?

Reply:
Do you mean the Azure plugin will break is that it?

Reply:
`toUriString` should return path with the scheme prefix, using just `toString` if you don't need that

Reply:
okay let me see if I can refactor accordingly

Reply:
@pditommaso this is a temporary fix for testing TES Azure, the ultimate goal is to include the account name in the AZ path only when appropriate

however now that I understand it's really the local path, likely we should implement a custom file copy strategy for it:
- `AzFileCopyStrategy`: current strategy which uses `azcopy` to transfer files
- `TesAzFileCopyStrategy`: custom strategy for TES Azure which simply the files mounted at `/<account-name>`

Reply:
The main problem is that TES Azure expects AZ paths in a different format from the Azure Batch executor

Reply:
so whats the recomendation?

Reply:
Agreed and we have to find a switch if its TES vs not